xmppserver/src/test/java/org/jivesoftware/openfire/spi/ConnectionTypeTest.java

```diff
@@ -16,110 +16,55 @@
-        final boolean result = input.isClientOriented();
-
-        // Verify result.
-        assertTrue(result);
+    @Parameterized.Parameters(name = "Verify that when ConnnectionType is \"{0}\" then isClientOriented returns \"{1}\"")
+	public static Iterable<Object[]> data() {
+		return Arrays.asList(new Object[][] {
+            {ConnectionType.SOCKET_S2S, false},
+            {ConnectionType.SOCKET_C2S, true},
+            {ConnectionType.BOSH_C2S, true},
+            {ConnectionType.WEBADMIN, true},
+            {ConnectionType.COMPONENT, false},
+            {ConnectionType.CONNECTION_MANAGER, false}
+		});
+	}
+
+	private final ConnectionType connType;
+    private final boolean expected;
+
+	public ConnectionTypeTest(ConnectionType connType, boolean expected){
+        this.connType = connType;
+        this.expected = expected;
     }
 
     /**
-     * Verifies that ConnectionType.BOSH_C2S is not determined to be 'client orientated'.
+     * Verifies that a given ConnectionType returns the appropriate 'client orientated' value.
      */
     @Test
-    public void testBoshC2SIsClientOrientated() throws Exception
+    public void testConnectionTypeHasCorrectIsClientOriented() throws Exception
     {
-        // Setup test fixture.
-        final ConnectionType input = ConnectionType.BOSH_C2S;
-
-        // Execute system under test.
-        final boolean result = input.isClientOriented();
-
-        // Verify result.
-        assertTrue(result);
-    }
-
-    /**
-     * Verifies that ConnectionType.WEBADMIN is not determined to be 'client orientated'.
-     */
-    @Test
-    public void testWebadminIsClientOrientated() throws Exception
-    {
-        // Setup test fixture.
-        final ConnectionType input = ConnectionType.WEBADMIN;
-
-        // Execute system under test.
-        final boolean result = input.isClientOriented();
-
-        // Verify result.
-        assertTrue(result);
-    }
-
-    /**
-     * Verifies that ConnectionType.COMPONENT is not determined to be 'client orientated'.
-     */
-    @Test
-    public void testComponentIsClientOrientated() throws Exception
-    {
-        // Setup test fixture.
-        final ConnectionType input = ConnectionType.COMPONENT;
-
-        // Execute system under test.
-        final boolean result = input.isClientOriented();
-
-        // Verify result.
-        assertFalse(result);
-    }
-
-    /**
-     * Verifies that ConnectionType.CONNECTION_MANAGER is not determined to be 'client orientated'.
-     */
-    @Test
-    public void testConnectionManagerIsClientOrientated() throws Exception
-    {
-        // Setup test fixture.
-        final ConnectionType input = ConnectionType.CONNECTION_MANAGER;
-
         // Execute system under test.
-        final boolean result = input.isClientOriented();
+        final boolean result = connType.isClientOriented();
 
         // Verify result.
-        assertFalse(result);
+        assertTrue(result == expected);
```

Comment:
Could you change this into `assertEquals(expected, result)` ?

Reply:
Good call. Got thrown by junit having no `assertEquals(boolean, boolean)` but the `(Object,Object)` variant works fine. When I change an expected result, the output looks much nicer.